packages/graphback-codegen-resolvers/src/plugins/ResolverGeneratorPlugin.ts

```diff
@@ -0,0 +1,86 @@
+import { getModelTypesFromSchema, GraphbackCoreMetadata, GraphbackPlugin, TypeScriptResolversVisitor } from '@graphback/core';
+import { buildASTSchema, GraphQLField, GraphQLFieldMap, GraphQLObjectType, GraphQLOutputType, GraphQLSchema, isListType, isNonNullType, parse, printSchema, visit } from 'graphql';
+import { ResolverGeneratorOptions } from '../api/ResolverGeneratorOptions';
+import * as pluralize from "pluralize";
+
+export interface ResolverDefinition {
+    generated: ResolverBaseTypeMap[]
+}
+
+export interface ResolverBaseTypeMap {
+    [typeName: string]: ResolverMap
+}
+
+export interface ResolverMap {
+    Query?: ResolverFunctionMap[]
+}
+
+export interface ResolverFunctionMap {
+    [name: string]: string;
+}
+
+export class ResolverGeneratorPlugin extends GraphbackPlugin {
+    private resolverDefinition: ResolverDefinition;
+    constructor(config?: ResolverGeneratorOptions) {
+        super();
```

Comment:
In my mind plugins will need to deal with writing files themselves. 
For example, resolver plugin will accept some layout config that specifies how to spread files etc.